tests/non-js-resources/subtests/service-worker/webpack/lib/sw-info-plugin.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const {
+  sources: { ConcatSource },
+  util: { createHash },
+  Compilation,
+} = require('webpack');
+
+/** @typedef {import('webpack/lib/Compiler')} Compiler */
+
+module.exports = class ServiceWorkerInfoPlugin {
+  constructor({
+    test = file => file === 'sw.js',
+    filterAssets = () => true,
+  } = {}) {
+    this.test = test;
+    this.filterAssets = filterAssets;
+  }
+
+  /** @param {Compiler} compiler */
+  apply(compiler) {
+    compiler.hooks.thisCompilation.tap(
+      'ServiceWorkerInfoPlugin',
+      compilation => {
+        compilation.hooks.processAssets.tap(
+          {
+            name: 'ServiceWorkerInfoPlugin',
+            stage: Compilation.PROCESS_ASSETS_STAGE_SUMMARIZE,
```

Comment:
It seems like this is executing too early - there are no HTML assets present in `assetsList` / `compilation.getAssets()`.

Reply:
Indeed. Upgrading html-webpack-plugin to major version 5 fixes that.